spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -272,6 +284,67 @@ public void deleteWhere(Filter[] filters) {
     }
   }
 
+  @Override
+  public StructType partitionSchema() {
+    Schema schema = icebergTable.spec().schema();
```

Comment:
we should probably get this from the Partitions Metadata table as well rather than converting it again here

Reply:
if don't converting, not the desired result. so
```
hours(ts):
    ts_hour=429536
    days(ts):
    requirement failed: Literal must have a corresponding value to date, but class Date found.
    months(ts):
    ts_month=588
    years(ts)
    ts_year=49
```